### PR TITLE
major code cleanup

### PR DESCRIPTION
**Pull request by paulfantom**
- to increase security binary files should be owned by root not by user who executes that file
- move user and group creation to `server` section as those aren't needed when only installing client
- remove hacky installation of packages (do not use `with_indexed_items`)
- rename `ansible_support_packages` to `minio_ansible_support_packages` as every variable starting from `ansible_` should come from ansible itself, not from role.
- remove `python_sni_pip_dependencies` and just list packages in task for better readability
- remove `systemd_units_dir` and always use `/etc/systemd/system` value as is recommended by systemd
- remove `initd_conf_dir` variable and only use its value for better readability

### CODE REVIEW

**Comment by SuperQ** [comment]
Do we want to add a cleanup for the old unit file location?

**Comment by paulfantom** [comment]
That's a good question. Adding this won't change anything in how it operates since files located in /etc/systemd/system takes priority over ones located in lib/systemd/system.
@atosatto WDYT?

**Comment by atosatto** [comment]
It's perfectly fine to stick with systemd unit files in `/etc/systemd/system`

File: tasks/client.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 ---
-
 - name: download minio client
   get_url:
     url: "{{ minio_client_download_url }}"
     dest: "{{ minio_client_bin }}"
-    owner: "{{ minio_user }}"
-    group: "{{ minio_group }}"
+    owner: "root"
```
Comment (atosatto) [inline tasks/client.yml]: This was done as a kind of "security measure".
Just not to download binaries from the internet as "root".
But I am not sure it still makes sense.
Thoughts?

Comment (paulfantom) [reply tasks/client.yml]: It isn't good to run any binary as a user who can edit this binary. We have a sane default for `minio_client_download_url` which is an official source, and we are using HTTPS so there shouldn't be any problem with tampering the file while transmitting to server.
I would also consider downloading binary once to local machine, validating checksum and then propagating it to target hosts. This increases security and reduces needed network bandwidth when used with more than 1 target. We are doing the same with most of the roles in @cloudalchemy :)

**Pull request merged**